jsettlers.mapcreator/src/jsettlers/mapcreator/main/MapCreatorApp.java

```diff
@@ -14,143 +14,189 @@
-			private static final long serialVersionUID = 648829725137437178L;
-
-			@Override
-			public Component getListCellRendererComponent(JList<?> list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
-				IMapDefinition map = (IMapDefinition) value;
-				String longMapId = map.getMapId();
-				String shortMapId = longMapId == null ? "no map id" : longMapId.substring(0, Math.min(longMapId.length(), 8));
-				String displayName = map.getMapName() + " \t   (" + shortMapId + ")      created: " + map.getCreationDate();
-				return super.getListCellRendererComponent(mapList, displayName, index, isSelected, cellHasFocus);
-			}
-		});
-		panel.add(new JScrollPane(mapList));
-
-		JButton button = new JButton("Open");
-		button.addActionListener(new ActionListener() {
-			@Override
-			public void actionPerformed(ActionEvent arg0) {
-				loadMap(mapList.getSelectedValue());
-			}
-		});
-		panel.add(button, BorderLayout.SOUTH);
 
-		return panel;
-	}
+				if (Boolean.parseBoolean(options.getProperty("delete-actionconfig"))) {
+					new File(actionconfig).delete();
+				}
 
-	private JPanel createNewMapPanel() {
-		JPanel panel = new JPanel();
-		panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
-		final MapHeaderEditor headerEditor = new MapHeaderEditor(DEFAULT, true);
-		panel.add(headerEditor);
-		headerEditor.setBorder(BorderFactory.createTitledBorder("Map settings"));
-
-		JPanel ground = new JPanel();
-		ground.setBorder(BorderFactory.createTitledBorder("Ground type"));
-		panel.add(ground);
-
-		final SpinnerListModel groundType = new SpinnerListModel(Arrays.asList(GROUND_TYPES));
-		JSpinner groundTypes = new JSpinner(groundType);
-		ground.add(groundTypes);
-
-		JButton createMapButton = new JButton("Create map");
-		panel.add(createMapButton);
-		createMapButton.addActionListener(new ActionListener() {
-			@Override
-			public void actionPerformed(ActionEvent arg0) {
-				new EditorWindow(headerEditor.getHeader(), ELandscapeType.valueOf(groundType.getValue().toString()));
-				close();
+				if ("new".equals(action)) {
+					// Sample property file:
+					// action=new
+					// map-name=OpenTest 1
+					// lanscape-type=GRASS
+					// width=100
+					// height=100
+					// min-player=1
+					// max-player=3
+
+					String mapName = prop.getMapName();
+					String description = prop.getMapDescription();
+					ELandscapeType lanscapeType = prop.getLanscapeType();
+					int width = prop.getWidth();
+					int height = prop.getHeight();
+					int minPlayer = prop.getMinPlayerCount();
+					int maxPlayer = prop.getMaxPlayerCount();
+					MapFileHeader header = new MapFileHeader(MapType.NORMAL, mapName, null, description, (short) width, (short) height,
+							(short) minPlayer,
+							(short) maxPlayer, null, new short[MapFileHeader.PREVIEW_IMAGE_SIZE * MapFileHeader.PREVIEW_IMAGE_SIZE]);
+
+					new EditorControl(header, lanscapeType);
+
+				} else {
+					String mapId = prop.getMapId();
```

Comment:
This is the workaround to start a second editor, because of the black window problem, but if you're change really fixes this problem, I will completely remove this code